test-app/tests/integration/components/combobox-test.gts

```diff
@@ -0,0 +1,854 @@
+
+    // NOTE: Our selected option is currently at the top of the list!
+    await render(<template>
+      <ComboboxControl
+        @selected="a"
+        @options={{options}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover by clicking it
+    await click('[data-combobox]');
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'PageDown');
+
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'true');
+
+    // Verify our first item is no longer "active"
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it makes the first option "active" when the PAGEUP key is pressed', async function (assert) {
+    let options = ['a', 'b', 'c', 'd', 'e', 'f'];
+
+    // NOTE: Our selected option is currently at the bottom of the list!
+    await render(<template>
+      <ComboboxControl
+        @selected="f"
+        @options={{options}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover by clicking it
+    await click('[data-combobox]');
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'PageUp');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    // Verify our last item is no longer "active"
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it makes the first option "active" when the HOME key is pressed', async function (assert) {
+    let options = ['a', 'b', 'c', 'd', 'e', 'f'];
+
+    // NOTE: Our selected option is currently at the bottom of the list!
+    await render(<template>
+      <ComboboxControl
+        @selected="f"
+        @options={{options}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover by clicking it
+    await click('[data-combobox]');
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'Home');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    // Verify our last item is no longer "active"
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  /**
+   * This test covers the case where we have a selected option,
+   * the user clears the input completely, and then the component's
+   * input tag is blurred.  In that case, we want to clear the input
+   * and reset their selection to null.
+   */
+  test('it calls `@onChange` with null after a user clears their original input selection', async function (assert) {
```

Comment:
Could just be me missing this...but is there a test that verifies that when a user focuses on the combobox input when text is already there, that it selects ALL test?

Reply:
I'm not sure if we can test this! Do you know how to test what is currently selected?  I was under the impression there's not a way to do this with qunit, but I could be wrong!

Reply:
- [ ] Let me try `document.getSelection().toString()` and see if that'll work!

Reply:
Good suggestion! That worked! Test added.